Exploration-of-the-Oppia-stack.md

```diff
@@ -0,0 +1,431 @@
+## Table of Contents
```

Comment:
Can we avoid having `exploration` word here? (As this can be confusing for contributors as we use this term in the codebase/app for some entity!)

Maybe: `Deep dive` or `walk-through`?

Reply:
I think "walk-through" works well. What do you think about "Tour of the Oppia stack"?

Reply:
Based on discussions with seanlip, I've turned this page into more of a tutorial on finding the code that needs to change to support a new feature